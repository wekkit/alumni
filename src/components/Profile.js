import React, { Component } from 'react'

class RepoLink extends Component {
  render() {
      return (
        <div className='card mx-auto'>
          <a href={this.props.project.deployedUrl} target='_blank' className='card-header'>{this.props.header}: {this.props.project.repoName}</a>
          {this.props.project.previewImage && <img src={this.props.project.previewImage} alt='Not found' className='img-thumbnail' />}
        </div>
      )
  }
}

class Profile extends Component {
  render() {
    return (
      <div className='card scroll-section' id='profile'>
        <div className='card-block'>
          <h1>{this.props.data.githubLogin}</h1>
          <hr/>
          <RepoLink project={this.props.data.project1} header='Project 1' />
          <RepoLink project={this.props.data.project2} header='Project 2' />
          <RepoLink project={this.props.data.project3} header='Project 3' />
        </div>
      </div>
    )
  }
}

export default Profile