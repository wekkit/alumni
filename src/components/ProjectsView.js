import React, { Component } from 'react'
import { IndexLink } from 'react-router'
import $ from 'jquery'

import * as data from '../data.json'
import Filters from './Filters'
import ProjectCard from './ProjectCard'

let duplicateCheck = []
const projectData = data.default.reduce((accum, elem) => {

        elem.project1.project = 1
        elem.project1.creator = elem.githubLogin
        accum.push(elem.project1)

        elem.project2.project = 2
        elem.project2.creator = elem.githubLogin
        accum.push(elem.project2)

        elem.project3.project = 3
        elem.project3.collaborators.push(elem.githubLogin)
        elem.project3.creator = elem.project3.collaborators.join(', ')
        let indexCheck = 0
        elem.project3.collaborators.forEach((e) => {
          duplicateCheck.forEach(check => {
            if (e === check) indexCheck++
          })
        })
        if (indexCheck === 0) {
          duplicateCheck = duplicateCheck.concat(elem.project3.collaborators)
          accum.push(elem.project3)
        }

        return accum
      }, [])

class ProjectsView extends Component {
  constructor(props) {
    super(props)
    this.state = {
      projectData: projectData,
      projectFilter: [],
      projectSearch: '',
      selectedProject: null
    }
  }

  filterHandler(e) {
    const newFilter = this.state.projectFilter
    const changedValue = parseInt(e.target.value, 0)
    const index = this.state.projectFilter.indexOf(changedValue)
    if (index >= 0) {
      newFilter.splice(index, 1)
    } else {
      newFilter.push(changedValue)
    }
    this.setState({
      projectFilter: newFilter
    })
    let filteredData = projectData.filter(data => {
      return data.repoName.toLowerCase().includes(this.state.projectSearch.toLowerCase())
    })
    for (let i = 0; i < this.state.projectFilter.length; i++) {
      filteredData = filteredData.filter((elem) => {
        if (elem.project === this.state.projectFilter[i]) return false
        else return true
      })
    }
    this.setState({
      projectData: filteredData
    })
  }

  searchHandler(e) {
    this.setState({ projectSearch: e.target.value })
    let filteredData = projectData.filter(data => {
      return data.repoName.toLowerCase().includes(e.target.value.toLowerCase())
    })
    for (let i = 0; i < this.state.projectFilter.length; i++) {
      filteredData = filteredData.filter((elem) => {
        if (elem.project === this.state.projectFilter[i]) return false
        else return true
      })
    }
    this.setState({ projectData: filteredData })
  }

  clickHandler(e) {
    const clickedProject = JSON.parse(e.target.id)
    if (clickedProject.project !==  3) {
      $.get('https://api.github.com/repos/'+ clickedProject.creator + '/' + clickedProject.repoName + '/readme?access_token=beb047469c874159724f9f479ab184ce22d9a164')
        .done((data) => {
          $.get(data.download_url)
            .done(readme => {
              clickedProject.readme = readme
              this.setState({
                selectedProject: clickedProject
              })
            })
        })
    } else {
      $.get('https://api.github.com/repos/'+ clickedProject.creator.split(',')[0] + '/' + clickedProject.repoName + '/readme?access_token=beb047469c874159724f9f479ab184ce22d9a164')
        .done((data) => {
          $.get(data.download_url)
            .done(readme => {
              clickedProject.readme = readme
              this.setState({
                selectedProject: clickedProject
              })
            })
        })
    }
  }

  render() {
    return (
      <div className='container'>
        <div className='row'>
          <div className='col-md-5 pane'>
            <Filters
              filterHandler={this.filterHandler.bind(this)}
              searchHandler={this.searchHandler.bind(this)}
              selectedProject={this.state.selectedProject}
            />
          </div>

          <div className='col-md-7 scroll-section'>
            <hr/>
            <h1>These are WDI7's projects.</h1>
            <IndexLink to='/' className='btn btn-primary'>View Developers</IndexLink>
            <hr/>

            {
              (this.state.projectSearch.length > 0) &&
                <p>Searching: {this.state.projectSearch}</p>
            }

            <div className='card-columns'>
              {
                this.state.projectData.map((elem, i) => {
                  return <ProjectCard
                            project={elem}
                            key={i}
                            clickHandler={this.clickHandler.bind(this)}
                          />
                })
              }
            </div>

          </div>
        </div>
      </div>
    )
  }

}

export default ProjectsView
