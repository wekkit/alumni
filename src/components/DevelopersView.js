import React, { Component } from 'react'
import * as data from '../data.json'
import Card from './Card'
import Profile from './Profile'
import '../App.css'
import '../bg.css'

class DevelopersView extends Component {
  constructor(props) {
    super(props)
    this.state = {
      data: data.default,
      filters: data.default.reduce((accum, e) => {
                  accum.push(e.githubLogin)
                  return accum
                }, []),
      users: data.default.map(e => {
                return e.githubLogin
              }),
      profileUser: data.default[0]
    }
  }

  cardClickHandler(e) {
    const clickedUser = data.default.filter(elem => {
        if (elem.githubLogin.toString().toLowerCase() === e.target.id.toString().toLowerCase()) return true
        else return false
      })[0]
    this.setState({
      profileUser: clickedUser
    })
  }

  render() {
    // <Filters data={this.state.users} changeHandler={this.filter.bind(this)} />
    return (
      <div className='container'>
        <div className='row'>
          <div className='col-md-7 scroll-section'>
            <hr/>
            <h1>This is WDI7.</h1>
            <hr/>
            <div className='card-deck'>
              {this.state.data.map((elem, i) => {
                return (
                  <Card key={i} person={elem} onClick={this.cardClickHandler.bind(this)}/>
                )
              })}
            </div>
          </div>

          <div className='col-md-5 pane'>
            <Profile data={this.state.profileUser} />
          </div>

        </div>
      </div>
    )
  }
}


export default DevelopersView;
