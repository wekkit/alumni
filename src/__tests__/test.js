import React from 'React'
import Link from 'react-router'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import * as data from '../data.json'

import Profile from '../components/Profile'
import DevelopersView from '../components/DevelopersView'
import ProjectsView from '../components/ProjectsView'

it('renders an empty profile correctly', () => {
  const tree = renderer.create(
    <Profile />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders with enzyme', () => {
  const devView = shallow(
    <DevelopersView data={data}/>
  )
  expect(devView.length).toBe(1)
})

it('has everyone\'s profile cards', () => {
  const devView = shallow(<DevelopersView />)
  expect(devView.find('DeveloperCard').length).toBe(15)
})

it('goes from developer view to project view', () => {
  const devView = shallow(<DevelopersView />)
  const projView = shallow(<ProjectsView />)
  devView.find('Link').simulate('click')
  expect(projView.length).toBe(1)
})

