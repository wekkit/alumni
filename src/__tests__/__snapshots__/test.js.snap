// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an empty profile correctly 1`] = `
<div
  className="card scroll-section"
  id="profile"
/>
`;
